app/Console/Commands/ExportEntityTypes.php

```diff
@@ -0,0 +1,563 @@
+            return 1;
+        }
+
+        try {
+            DB::connection(static::$destination)->getPdo();
+        } catch(\Exception $e) {
+            $this->print("Could not connect to'". static::$destination . "' database " . $e->getMessage());
+            return 1;
+        }
+
+        $this->startPrinter();
+        try {
+            $sourceConfig = DB::connection(static::$source)->getConfig();
+            $this->printConnectionConfig("Source Connection -> " . static::$source, $sourceConfig);
+
+            $destinationConfig = DB::connection(static::$destination)->getConfig();
+            $this->printConnectionConfig("Destination Connection -> ". static::$destination , $destinationConfig);
+
+            $entityId = $this->option('entity');
+
+            $entity = Entity::find($entityId);
+            if(!$entity){
+                $this->print('Entity not found');
+                return 1;
+            }
+
+            $this->print("Analyze structure starting at base entity: " . $entity->name . " (" . $entity->id . ")");
+            $this->printSectionHeading("Traverse Entity Tree & Extracting Data Models");
+
+            $data = $this->collectAllDataFromEntityChildren($entity);
+
+            $this->printReceipt($data);
+            $this->print("Beginning Transaction");
+
+            DB::connection(static::$destination)->beginTransaction();
+            $user = $this->selectOrCreateTransferUser();
+            $this->transferData($user, $data);
+
+            $this->print("Waiting for user confirmation of this receipt ...");
+            if($this->confirm('Check the receipt carefully before proceeding')) {
+                $this->print("User confirmed receipt. Proceeding with commit");
+                DB::connection(static::$destination)->commit();
+                // This path did not return anything. I return 0 here (see bash return codes)
+                return 0;
+            } else {
+                $this->print("User declined receipt. Rolling back transaction and exiting Program");
+                DB::connection(static::$destination)->rollBack();
+                return 1;
+            }
+        } catch(Exception $e) {
+            $this->print("The program failed unexpectedly:\n" . $e->getMessage() . "\n" . $e->getTraceAsString());
+            return 1;
+        } finally {
+            $this->print("Program finished successfully\nThank you for using this tool\nHave a nice day!");
+            $this->closePrinter();
+            // This path did not return anything. I return 0 here (see bash return codes)
+            return 0;
+        }
+    }
+
+    private function extractThesaurusConcepts($model, array &$allConcepts): void {
+        $collection = ThConcept::where('concept_url', $model->thesaurus_url)->get();
+        foreach($collection as $concept) {
+            $allConcepts[$model->thesaurus_url] = $concept;
+        }
```

Comment:
Does it make sense to iterate over an array here?

1. There should be only `concept_url` matching the model's `thesaurus_url` in the DB (`concept_url` is unique in `th_concept`-table)
2. Looping over `collection` and setting `allConcepts[$model->thesaurus_url]` will only set the content of the last item or am I missing something?